src/Microsoft.AspNetCore.Mvc.Razor.Host/IMvcRazorHostWithCodeGenerationContext.cs

```diff
@@ -0,0 +1,20 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Razor;
+using Microsoft.AspNetCore.Razor.CodeGenerators;
+
+namespace Microsoft.AspNetCore.Mvc.Razor
+{
+    /// <summary>
+    /// Specifies the contracts for a Razor host that parses Razor files and generates C# code.
+    /// </summary>
+    public interface IMvcRazorHostWithTemplateEngineContext
```

Comment:
@Eilon as discussed we'd end up with a fork in the code when we see this interface. Feel free to come up with a better name,